src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourceDictionary.cs

```diff
@@ -1771,35 +1794,55 @@ private void NotifyOwners(ResourcesChangeInfo info)
         /// </summary>
         private void ValidateDeferredResourceReferences(object resourceKey)
         {
-            if (_deferredResourceReferences is null)
+            if (FrameworkAppContextSwitches.DisableDynamicResourceOptimization)
             {
-                return;
-            }
-
-            if (resourceKey is null)
-            {
-                foreach (DeferredResourceReference deferredResourceReference in _deferredResourceReferences)
+                if (_weakDeferredResourceReferences != null)
                 {
-                    Inflate(deferredResourceReference);
+                    foreach (Object o in _weakDeferredResourceReferences)
+                    {
+                        DeferredResourceReference deferredResourceReference = o as DeferredResourceReference;
+                        if (deferredResourceReference != null && (resourceKey == null || Object.Equals(resourceKey, deferredResourceReference.Key)))
+                        {
+                            // This will inflate the deferred reference, causing it
+                            // to be removed from the list.  The list may also be
+                            // purged of dead references.
+                            deferredResourceReference.GetValue(BaseValueSourceInternal.Unknown);
+                        }
+                    }
                 }
             }
             else
             {
-                DeferredResourceReference deferredResourceReference = _deferredResourceReferences.Get(resourceKey);
+                if (_deferredResourceReferencesList is null)
+                {
+                    return;
+                }
 
-                Inflate(deferredResourceReference);
-            }
+                if (resourceKey is null)
+                {
+                    foreach (DeferredResourceReference deferredResourceReference in _deferredResourceReferencesList)
+                    {
+                        Inflate(deferredResourceReference);
+                    }
+                }
+                else
+                {
+                    DeferredResourceReference deferredResourceReference = _deferredResourceReferencesList.Get(resourceKey);
 
-            return;
+                    Inflate(deferredResourceReference);
+                }
 
-            void Inflate(DeferredResourceReference deferredResourceReference)
-            {
-                if (deferredResourceReference is not null)
+                return;
+
+                void Inflate(DeferredResourceReference deferredResourceReference)
```

Comment:
nit - Indentation for this method seems off. could you please fix it.

Reply:
It seems off, but it is correct